controlplane/kubeadm/internal/controllers/controller_test.go

```diff
@@ -753,7 +756,10 @@ func TestKubeadmControlPlaneReconciler_adoption(t *testing.T) {
 			managementClusterUncached: fmc,
 		}
 
-		g.Expect(r.reconcile(ctx, cluster, kcp)).To(Equal(ctrl.Result{}))
+		_, hasAdopted, err := r.initControlPlaneScope(ctx, cluster, kcp)
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(hasAdopted).To(BeTrue())
```

Comment:
hasAdopted seems very misleading here. Basically the return parameter is that there are adoptable machines and if we called adoptMachines. It doesn't mean that we actually adopted a machine

Would be good to clarify here a bit (e.g. by renaming the bool) and I would suggest to add a note to the godoc of initControlPlaneScope that this bool means there are Machines to be adopted, not that they have been adopted.

e.g. I would rename machineAdopted (in controller.go) and hasAdopted here to something like adoptableMachinesExist (also in scale_test.go)